impl/README.md

```diff
@@ -0,0 +1,188 @@
+
+### Quick version
+
+For a quick introduction, we will use a simple workflow [definition](../examples/simpleGet/src/main/resources/get.yaml) that performs a get call. 
+We are going to show two ways of invoking the workflow: 
+  - blocking the thread till the get request goes through
+  - returning control to the caller, so the main thread continues while the get is executed
+
+In order to execute the workflow, blocking the thread till the HTTP request is completed, you should write
+
+``` java 
+try (WorkflowApplication appl = WorkflowApplication.builder().build()) {
+      logger.info(
+          "Workflow output is {}",
+          appl.workflowDefinition(WorkflowReader.readWorkflowFromClasspath("get.yaml"))
+              .instance(Map.of("petId", 10))
+              .start()
+              .join());
+    }
+```
+You can find the complete java code [here](../examples/simpleGet/src/main/java/BlockingExample.java)
+
+In order to execute the workflow without blocking the calling thread till the HTTP request is completed, you should write
+
+``` java 
+  try (WorkflowApplication appl = WorkflowApplication.builder().build()) {
+      appl.workflowDefinition(WorkflowReader.readWorkflowFromClasspath("get.yaml"))
+          .instance(Map.of("petId", 10))
+          .start()
+          .thenAccept(node -> logger.info("Workflow output is {}", node));
+    }
+```
+When the HTTP request is done, both examples will print a similar output
+
+
+```shell
+Workflow output is {"id":10,"category":{"id":10,"name":"string"},"name":"doggie","photoUrls":["string"],"tags":[{"id":10,"name":"string"}],"status":"string"}
+
+You can find the complete java code [here](../examples/simpleGet/src/main/java/NotBlockingExample.java)
+
+### Detailed version
+
+To discuss runtime API we are going to use a couple of workflow:
+- [listen.yaml](../examples/events/src/main/listen.yaml), which waits for an event reporting a temperature greater than 38
+- [emit.yaml](../examples/events/src/main/emit.yaml), which emits events with a certain temperature, specified as workflow parameter.
+
+Here is a summary of what we are trying to do: We will start the `listen.yaml` workflow, which will be completed when it receives an event with the proper temperature. It won't block the main thread while waiting for it. Then, we will send an event with a lower temperature that will be ignored. Finally, we will send an event with a higher temperature to complete the waiting workflow. 
+
+The first step is to create a [WorkflowApplication](core/src/main/java/io/serverlessworkflow/impl/WorkflowApplication.java) instance. An application is an abstraction that allows customization of different aspects of the workflow execution (for example, change the default `ExecutorService` for thread spawning)
+
+Since `WorkflowApplication` implements `Autocloseable`, we better use a try...finally block, ensuring any resource that the workflow might have used is freed when done. 
```

Comment:
use try-with-resources term, as it's more accurate
```suggestion
Since `WorkflowApplication` implements `Autocloseable`, we better use a **try-with-resources** block, ensuring any resource that the workflow might have used is freed when done. 
```